### PR TITLE
batched patches_grid

### PR DESCRIPTION
**Pull request by AnihilatorGun**

### CODE REVIEW

**Comment by AnihilatorGun** [comment]
Mb some comments required?

File: dpipe/im/grid.py [hunk 1]
```diff
@@ -50,6 +50,27 @@ def get_boxes(shape: AxesLike, box_size: AxesLike, stride: AxesLike, axis: AxesL
...
+def make_batch(divide_iterator: Iterable, batch_size: int = 1):
+    patches_to_batch = []
+    n = 0
+    for patch in divide_iterator:
+        patches_to_batch.append(torch.from_numpy(patch))
+        n += 1
+        
+        if n == batch_size:
+            n = 0
+            yield torch.cat(patches_to_batch).numpy()
```
Comment (vovaf709) [inline dpipe/im/grid.py:62]: Why u need numpy -> torch -> numpy conversions?

File: dpipe/im/grid.py [hunk 2]
```diff
@@ -50,6 +50,27 @@ def get_boxes(shape: AxesLike, box_size: AxesLike, stride: AxesLike, axis: AxesL
         yield make_box_([start, np.minimum(start + box_size, shape)])
 
 
+def make_batch(divide_iterator: Iterable, batch_size: int = 1):
```
Comment (vovaf709) [inline dpipe/im/grid.py:53]: Do u really need defining this function? Wouldn't https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.batched be enough with casting to numpy?

File: dpipe/im/grid.py (hunk 1 above)
Comment (AnihilatorGun) [reply dpipe/im/grid.py:62]: Torch works faster than numpy, but inference_step expects numpy.ndarray, not torch.Tensor, so it is

File: dpipe/itertools.py [hunk 3]
```diff
@@ -124,7 +124,7 @@ def __init__(self, func: Callable, iterable: Iterable, *args, **kwargs) -> None:
         self.__args = args
         self.__kwargs = kwargs
 
-        self.__result_queue = Queue()
+        self.__result_queue = Queue(1)
```
[the remark this answers is not in the document]
Comment (AnihilatorGun) [reply dpipe/itertools.py:127]: It works faster for urolithiasis + memory limitation

File: dpipe/im/grid.py (hunk 2 above)
Comment (AnihilatorGun) [reply dpipe/im/grid.py:54]: Done in new commit, now use batched

**Comment by AnihilatorGun** [comment]
Could I add `more_itertool` to requirements? I thought it is installed by default

File: dpipe/itertools.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vovaf709) [reply dpipe/itertools.py:127]: Hmm, do you understand why it is faster? As I understand `Queue(1)` means that prediction thread will wait until `__next__` is called. But `Queue()` means that prediction thread doesn't wait for anything, so seems like it should be faster

**Comment by vovaf709** [comment]
> Could I add `more_itertool` to requirements? I thought it is installed by default

Yep

File: dpipe/itertools.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AnihilatorGun) [reply dpipe/itertools.py:127]: GIL gives controll to main thread (combine.update) when queue.put is blocked. Here speed up is achieved

**Pull request merged**